### ISSUE TITLE
composer create-project sulu/skeleton fails when clearing cache

### ISSUE THREAD

**Issue by CVogel** [sulu/sulu#5889]
| Q | A
| --- | ---
| Bug? | yes
| New Feature? | no
| Sulu Version | 2.2.5
| PHP/Composer Version | 7.4.16 / 2.0.11 and 7.4.13 / 2.0.6

#### Actual Behavior

When running composer create-project sulu/skeleton my-project 
post install command cache:clear fails with exception

#### Expected Behavior

composer create-project sulu/skeleton runs without exception

#### Steps to Reproduce

1. Run `composer create-project sulu/skeleton my-project`
```
$ composer create-project sulu/skeleton
Creating a "sulu/skeleton" project at "./skeleton"
Installing sulu/skeleton (2.2.5)
  - Downloading sulu/skeleton (2.2.5)
  - Installing sulu/skeleton (2.2.5): Extracting archive
Created project in /tmp/skeleton
> @php -r "file_put_contents('.env.local', 'APP_ENV=dev' . PHP_EOL);"
> @php -r "file_put_contents('.env', str_replace('APP_SECRET=', 'APP_SECRET=' . bin2hex(random_bytes(16)), file_get_contents('.env')));"
Loading composer repositories with package information
Updating dependencies
Lock file operations: 147 installs, 0 updates, 0 removals
  - Locking antishov/doctrine-extensions-bundle (v1.4.3)
  - Locking behat/transliterator (v1.3.0)
... output continues

```

2. post install command cache:clear fails with following exception
```
...
  - Installing gedmo/doctrine-extensions (v3.0.3): Extracting archive
...
  - Configuring symfony/web-profiler-bundle (>=3.3): From github.com/symfony/recipes:master
Executing script cache:clear [KO]
 [KO]
Script cache:clear returned with error code 1
!!  
!!   // Clearing the cache for the dev environment with debug                       
!!   // true                                                                        
!!  
!!  
!!  In TreeListener.php line 71:
!!                                                                                 
!!    Tree object class: Sulu\Bundle\MediaBundle\Entity\Collection must have tree  
!!     metadata at this point                                                      
!!                                                                                 
!!  
!!  cache:clear [--no-warmup] [--no-optional-warmers]
!!  
!!  
!!   // Clearing the cache for the dev environment with debug                       
!!   // true                                                                        
!!  
!!  
!!  In FileLoader.php line 173:
!!                                                                                 
!!    Tree object class: Sulu\Bundle\MediaBundle\Entity\Collection must have tree  
!!     metadata at this point in /tmp/skeleton/vendor/sulu/sulu/src/Sulu/Bundle/C  
!!    ontactBundle/Resources/config/routing_api.yml (which is being imported from  
!!     "/tmp/skeleton/config/routes/sulu_admin.yaml"). Make sure there is a loade  
!!    r supporting the "rest" type.                                                
!!                                                                                 
!!  
!!  In TreeListener.php line 71:
!!                                                                                 
!!    Tree object class: Sulu\Bundle\MediaBundle\Entity\Collection must have tree  
!!     metadata at this point                                                      
!!                                                                                 
!!  
!!  cache:clear [--no-warmup] [--no-optional-warmers]
!!  
!!  
Script @auto-scripts was called via post-update-cmd
```

#### Possible Solutions
I believe this exception happens because of the gedmo/doctrine-extensions version 3.*.

I can "fix" this exception afterwards with the following steps
* delete composer.lock `rm composer.lock`
* clear vendor folder `rm -rf vendor/*``
* require version v2.4.42 of gedmo/doctrine-extension explicitly this will run composer install automatically afterwards `composer require gedmo/doctrine-extensions v2.4.42`

I found the merged pull request to "allow newer version of gedmo doctrine extensions" https://github.com/sulu/sulu/pull/5753

And I'm afraid that the XML Doctrine Mapping that is used in the MediaBundle might not be supported anymore with version 3 of the extension. But I'm just guessing.

I'm happy to create an issue in the sulu/sulu repository but I wanted to leave my "fix" here and I wasn't sure if this exception only occurs on new installs.

Warm regards

Christian

**Comment by chirimoya** [comment]
@alexander-schranz `composer require doctrine/doctrine-bundle 2.2.4` works for me. Maybe related to https://github.com/doctrine/DoctrineBundle/pull/1304

**Comment by alexander-schranz** [comment]
I tried #1304 but sadly this PR does not fix our issue with the gedmo tree extension metadata not loaded.

**Comment by alexander-schranz** [comment]
Current workaround would be to adjust the `composer.json` in your project like shown below. Unfortunately, there is not only an issue with doctrine/doctrine-bundle but also an issue with jms/serializer-bundle:

```json
    "conflict": {
        "doctrine/doctrine-bundle": "2.3.0",
        "jms/serializer-bundle": "3.9.0"
    }
```

```bash
composer update
```

**Comment by Dekminoz** [comment]
> Current workaround would be to adjust the `composer.json` in your project like shown below. Unfortunately, there is not only an issue with doctrine/doctrine-bundle but also an issue with jms/serializer-bundle:
> 
> ```json
>     "conflict": {
>         "doctrine/doctrine-bundle": "2.3.0",
>         "jms/serializer-bundle": "3.9.0"
>     }
> ```
> 
> ```shell
> composer update
> ```

Thank you, I’ve been having this problem all morning. Now it's working

**Comment by CVogel** [comment]
Thank you for your quick response and analysis. I'm now also using the conflict section in composer.json.

**Comment by niklasnatter** [comment]
Thanks for reporting! This was fixed by #5891. Make sure to set the version of the `sulu/sulu` package to `~2.2.6` in the `composer.json` of your project to install the newest version.
